pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md

```diff
@@ -17,25 +16,66 @@ This section describes how to upgrade your Kommander Management cluster and all
+
+### Load the Docker images into your Docker registry (skip this section if you are running in an **air-gapped environment**)
+
+1.  Download the image bundle file:
+
+    ```bash
+    wget "https://downloads.mesosphere.com/kommander/airgapped/v2.2.0/kommander_image_bundle_v2.2.0_linux_amd64.tar" -O kommander-image-bundle.tar
+    ```
+
+1.  Place the bundle in a location where you can load and push the images to your private Docker registry.
+
+1.  Ensure you set the `REGISTRY_URL` and `AIRGAPPED_TAR_FILE` variable appropriately, then use the following script to load the air-gapped image bundle:
+
+    ```bash
+    #!/usr/bin/env bash
+    set -euo pipefail
+    IFS=$'\n\t'
+
+    readonly AIRGAPPED_TAR_FILE=${AIRGAPPED_TAR_FILE:-"kommander-image-bundle.tar"}
+    readonly REGISTRY_URL=${REGISTRY_URL?"Need to set REGISTRY_URL. E.g: 10.23.45.67:5000"}
+
+    docker load <"${AIRGAPPED_TAR_FILE}"
+
+    while read -r IMAGE; do
+        echo "Processing ${IMAGE}"
+        REGISTRY_IMAGE="$(echo "${IMAGE}" | sed -E "s@^(quay|gcr|ghcr|docker).io@${REGISTRY_URL}@")"
+        docker tag "${IMAGE}" "${REGISTRY_IMAGE}"
+        docker push "${REGISTRY_IMAGE}"
+    done < <(tar xfO "${AIRGAPPED_TAR_FILE}" "index.json" | grep -oP '(?<="io.containerd.image.name":").*?(?=",)')
+    ```
+
+Based on the network latency between the environment of script execution and the docker registry, it can take a while to upload all the images to your image registry.
 
 ## Upgrade Kommander
 
 Before running the following command, ensure that your `dkp` configuration **references the Kommander Management cluster**, otherwise it attempts to run the upgrade on the bootstrap cluster. You can do this by setting the `KUBECONFIG` environment variable [to the appropriate kubeconfig file's location][k8s-access-to-clusters].
 
-<p class="message--note"><strong>NOTE:</strong> An alternative to initializing the KUBECONFIG environment variable as stated earlier is to use the <code>–kubeconfig=cluster_name.conf</code> flag. This ensures that Kommander is upgraded on the workload cluster.</p>
+<p class="message--note"><strong>NOTE:</strong> An alternative to initializing the KUBECONFIG environment variable as stated earlier is to use the <code>–kubeconfig=cluster_name.conf</code> flag. This ensures that Kommander upgrades on the workload cluster.</p>
+
+1.  Use the DKP CLI to upgrade Kommander and all the Platform Applications in the Management Cluster:
+
+    For air-gapped:
+
+    ```bash
+    dkp upgrade kommander --charts-bundle bundle.tar.gz
+    ```
 
-1.  Use the DKP CLI to upgrade Kommander and all the Platform Applications in the Management cluster:
+    For non air-gapped:
 
     ```bash
     dkp upgrade kommander
     ```
 
-    <A confirmation message appears once upgrade is complete.>
+    <--! A confirmation message appears once upgrade is complete.--! >
```

Comment:
@gracedo , do we have an update on this?

Reply:
If the error happened, the upgrade process is going to be terminated. Otherwise the CLI will just finish with `0` exist status. 

Do mean here adding an extra message that will confirm that the upgrade succeeded as well?

Reply:
@vespian we need to add the output here. AND we need to tell them how to troubleshoot if this doesn't work. Like what would a failure output look like? then what are the steps to fix that?

Reply:
Comman exists with non-zero exit status (in this case with `1`)